roles/keycloak/templates/codecentric-keycloak.yml.j2

```diff
@@ -75,6 +75,10 @@ extraVolumes: |
         secretName: {{ '{{ include "keycloak.fullname" . }}' }}-tls-certs
     - name: updated-cacerts
       emptyDir: {}  # work dir for update-truststore.sh
+    - name: certs-dir
```

Comment:
I am wondering about 2 issues:
1) Usually in K8 certs are always handled as secrets and not as cm. I understand that there is no secret in the CA certs, but there is some common practice established
2) `hostPath` is looking very weird to me. If you want to mount configMap you do it like https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#populate-a-volume-with-data-stored-in-a-configmap For secrets it is a bit different but is shown few lines above. Hostpath is used when you want to mount physical folder of the K8 worker node into the pod. It is bad in that sense, that the pod cannot be rescheduled to another node since the data is not there.

Reply:
1. Yes, certs are not mounted as configmaps anymore. 
2. That's what I wanted, as the ansible playbook is executed from inside a container, it mounts a physical folder from the manager node on the initContainer only, that containers loads the certs into the truststore and the volume is no longer needed.

Reply:
ehh, this is pretty problematic. From the Ansible<=>K8 point of view you should never make an assumption that you execute playbook on the same host as k8 (hostPath may be on a different host) since it may be easily changed and all of a sudden you get absolutely insane bugs not understanding why. I do not think saving one CM/Secret is worth of introducing bugs potential. It is just not what you do typically when provisioning applications to K8.
Imagine another use case where you deploy your applications using ArgoCD or similar stuff - it just runs somewhere and so you setup should be always referring to CMs/Secrets and eventually PVCs populated during init.

Reply:
Then WHAT needs to be implemented here? The volume approach might cause problems, the CM approach where the operator has to write the CA's wont work either. So... any ideas?

Reply:
Why does the CM approach not work? Maybe I missed something in the discussion. In my eyes the most correct approach is that inventory with which Ansible is triggered should have variable populated with the content of the certs (or it points to the local dir, but then the playbooks and not the role is responsible in reading this in) and those are used to create K8 Secret (or CM, but Secret is more logical). This Secret/CM is mounted as FS into the init container and it reads it in.

Reply:
>     2. That's what I wanted, as the ansible playbook is executed from inside a container, it mounts a physical folder from the manager node on the initContainer only, that containers loads the certs into the truststore and the volume is no longer needed.

I guess we have a misunderstanding here: Ansible role (what you are changing) is not a playbook like you say. Role is ONLY a sequence of steps (tasks) applied to the target. Playbook in difference is a mapping of tasks to targets (called plays). So inside the role you actually do not do any assumptions on from where it is executed (there are very rare exceptions to that, but that is advanced), you only describe what to do with the target.

Roles are designed to have access to variables passed to it (by the playbook applying this role to the target), own role defaults and role files/templates (which are "transferred" to the target during execution)

Reply:
Well, for me it's enough if someone can explain me where should I put my stuff to work, so I don't have to expend a crazy amount of time trying to figure it out looking into 3 different repos. I only want SSL from Keycloak to work. 

I guess that leave us with some jinja2 template for writing all the certificates into a CM and mount that into the initContainer. Let's hope I can figure it out how to do it and where to place it.

Reply:
New version, now a configmap is generated from a j2 template and the files from a configurable directory